pages/api/email/util/sendEmail.ts

```diff
@@ -48,15 +54,57 @@ export async function sendEmail (
   return [messagesSendResult.status, messagesSendResult.message]
 }
 
-function constructRecipientVariables (messages: Message[], signature: string) {
+function formatAllRecipents (messages: Message[]) {
+  const allRecipents: string[][] = []
+  messages.forEach((msg) => {
+    const ccRecipents = msg.cc
+    const bccRecipents = msg.bcc
+    const recipentMessage: string[] = []
+    recipentMessage.push(msg.to)
+    ccRecipents.forEach((recipient) => {
+      recipentMessage.push(recipient)
+    })
+    bccRecipents.forEach((recipient) => {
+      recipentMessage.push(recipient)
+    })
+    allRecipents.push(recipentMessage)
+  })
+  return allRecipents.flat(1)
+}
+
+function constructRecipientVariables (
+  messages: Message[],
+  signature: string) {
   const recipientVariables: { [email: string]: any } = {}
   messages.forEach((message) => {
+    const ccRecipents = message.cc
+    const bccRecipents = message.bcc
     recipientVariables[message.to] = {
       subject: message.subject,
       content: message.content +
         '<br/><br/>' +
         signature.replace('/assets/hbplogo.png', 'cid:hbplogo.png')
     }
+    if (ccRecipents) {
+      ccRecipents.forEach((recipient) => {
+        recipientVariables[recipient] = {
+          subject: message.subject,
+          content: message.content +
+        '<br/><br/>' +
+        signature.replace('/assets/hbplogo.png', 'cid:hbplogo.png')
+        }
+      })
+    }
+    if (bccRecipents) {
+      bccRecipents.forEach((recipient) => {
+        recipientVariables[recipient] = {
+          subject: message.subject,
+          content: message.content +
+        '<br/><br/>' +
+        signature.replace('/assets/hbplogo.png', 'cid:hbplogo.png')
+        }
+      })
```

Comment:
these two if statements for cc and bcc recipients look the same. Is there a way to abstract it? (i honestly dunno hehe)

Reply:
i could put it intoa function but that would slightly redundant since the code really is not doing much to warrant its own function